hfile.c

```diff
@@ -615,6 +615,47 @@ static hFILE *hopen_fd(const char *filename, const char *mode)
     return NULL;
 }
 
+static hFILE *hpreload_fd(const char *filename, const char *mode)
+{
+    if(!strchr(mode, 'r'))
+    {
+        return NULL;
+    }
+    
+    hFILE_fd *fp = NULL;
+    FILE *file = fopen(filename, mode);
+    if (!file) goto error;
+
+    fseek(file, 0, SEEK_END);
```

Comment:
You need to check the return value of fseek, incase someone has given a non-seekable stream to this function. (Eg /dev/stdin)